### PR TITLE
show applicable error/warning for expired or soon to be expired auth token

### PR DESCRIPTION
**Pull request by idodod**
In case a valid but expired token is used to login/authenticate, the error message will indicate so for the user.
Example:
`Error: failed executing ping request: auth token expired`
In case the token expiration is 14 days or under, it will display a warning stating when the token will expire, and then continue.
Example:
`Warning: token will expire in 1 day 9 hours`

### CODE REVIEW

File: cloud/auth.go [hunk 1]
```diff
@@ -562,6 +572,16 @@ func (c *Client) getSSHCredentials(challenge string, key *agent.Key) (credential
 	return credentials, nil
 }
 
+func (c *Client) maybeWarnAuthExpiration() {
+	if authExpirationWarningDisplayed || c.lastAuthMethodExpiry.IsZero() {
+		return
+	}
+	if diff := c.lastAuthMethodExpiry.Sub(time.Now().UTC()); diff >= 0 && diff <= authExpirationWarningGracePeriod {
```
Comment (alexcb) [inline cloud/auth.go:579]: `diff` doesn't make it clear if it's the duration until or after expiry.

maybe call it something like `remaining` instead?

```suggestion
	if remaining := c.lastAuthMethodExpiry.Sub(time.Now().UTC()); remaining >= 0 && remaining <= authExpirationWarningGracePeriod {
```

**Review by alexcb — APPROVED** [review]

File: cloud/auth.go [hunk 2]
```diff
@@ -14,11 +14,19 @@ import (
...
 	"github.com/earthly/earthly/util/fileutil"
+
+	"github.com/hako/durafmt"
 	"github.com/pkg/errors"
 	"golang.org/x/crypto/ssh"
 	"golang.org/x/crypto/ssh/agent"
 )
 
+const (
+	authExpirationWarningGracePeriod = time.Hour * 24 * 14
```
Comment (idodod) [inline cloud/auth.go:25]: @vladaionescu I set the warning to 14 days or under. Are you good with that or should I set it to 30 days or something else?

Comment (vladaionescu) [reply cloud/auth.go:25]: 14 days is reasonable!

Comment (idodod) [reply cloud/auth.go:25]: thanks for the quick reply :)

**Pull request merged**